lib/rouge/lexers/perl6pod.rb

```diff
@@ -0,0 +1,141 @@
+# -*- coding: utf-8 -*- #
+
+module Rouge
+  module Lexers
+    class Pod6 < RegexLexer
+      title "Perl 6 Pod"
+      desc "Perl 6 Plain' Ol Documentation (perl6.org)"
+
+      tag 'perl6pod'
+      aliases 'pod6'
+
+      filenames '*.pod', '*.pod6'
+
+      formatting_tokens = {
+        "B" => Generic::Strong,
+        "C" => Generic::Output,
+        "E" => Literal::Number::Other,
+        "I" => Generic::Emph,
+        "K" => Generic::Inserted,
+        "L" => Other,
+        "N" => Comment::Special,
+        "T" => Generic::Output,
+        "U" => Generic,
+        "Z" => Comment,
+      }
+
+      state :root do
+        rule(/\=begin pod/m, Keyword, :pod6)
+        rule(/^#.*$/, Comment)
+        rule(/./m, Text)
+      end
+
+      state :pod6 do
+        rule(/\=end pod/m, Keyword, :pop!)
+
+        rule(/\=(?:NAME|AUTHOR|VERSION|TITLE|SUBTITLE)/, Keyword, :semantic)
+
+        rule(/\=begin code/, Keyword, :block_code)
+        rule(/\=begin input/, Keyword, :block_input)
+        rule(/\=begin output/, Keyword, :block_output)
+        rule(/\=begin/, Keyword, :block)
+
+        rule(/\=head(\d+)\s+/, Keyword, :head)
+        rule(/\=item(\d+)\s+/, Keyword, :item)
+        rule(/\=input/, Keyword, :input)
+        rule(/\=output/, Keyword, :output)
+        rule(/\=defn/, Keyword)
+
+        rule(/(BCEIKLNTUZ)<([^>]*)>/) do |m|
```

Comment:
Is `(BCEIKLNTUZ)` meant to be a character class?

Reply:
Yes, it was, so I tried to fix that. However, this particular `rule` seemed to drop the `<` and `>` characters, which are important to keep. If I can get pointers on how to do something like this that keeps the `<` and `>` characters, I can put a working variant back into the PR.

I was wondering if there's a way to forcefully insert a character into the resulting document, so I can simply push the `<` and `>` back in. If there is such a method, I could rewrite some of the `=begin ... =end` logic to be a single block as well, I think. This would reduce the overall size of the codebase and duplicated efforts for all the special blocks that I have right now.

Reply:
You might want to read through Ruby's [`Regexp` documentation](https://ruby-doc.org/core-2.6.3/Regexp.html). It sounds like that might be helpful in consolidating a few things.

To answer your specific question, I think what you want to use are capture groups. This allows you to 'capture' the value matching the capture group's pattern. Using this, you could rewrite this:

```ruby
rule(/\=begin code/, Keyword, :block_code)
rule(/\=begin input/, Keyword, :block_input)
rule(/\=begin output/, Keyword, :block_output)
```

as this:

```ruby
rule(/(\=begin)( )(code|input|output)/i) do |m|
  s = case |m[3].downcase| # use the result of the third capture group
      when "code" then :block_code
      when "input" then :block_input
      when "output" then :block_output
  end
  groups Keyword, Text, Keyword # use the groups method to assign tokens to capture groups in order
  push s # push the new state onto the stack
end
```

The `rule` method can take a block (as in the example above) that accepts a single variable (the convention in Rouge is to call the variable `m` for 'match'). Within the block you can access the capture groups with the `[]` operator. `m[0]` is the complete pattern, `m[1]` is the first capture group, `m[2]` is the second capture group and so on.